### ISSUE TITLE
Better debounce

### ISSUE THREAD

**Issue by DJsushi** [thecode/ha-rpi_gpio#74]
Today, I put a binary sensor on my door that I made myself and connected it to the raspberry pi's GPIO. I really like this integration although I have a little issue with the debounce feature. Basically, right now, the debounce also creates a delay after the door is closed. Closing some doors is very difficult (especially when it's night and everybody is sleeping). When closing this door, there's a decent amount of bouncing. Now this what I have a problem with. The debounce feature doesn't immediately tell me the door is closed if I have a debounce of 2 seconds. It waits for 2 seconds, and THEN tells me the door closed. When I suggest is the option to change the debounce algorithm. I'd like to have a debounce algorithm which would tell me the door is closed immediately after the first contact of the binary sensor. And if there's some bouncing while closing the door, ignore that bouncing. This is important because for example I want to set my music volume immediately to lower and not risk 2-3 false triggers while the door is closing (2-3 false door closings and opening, when in reality it was just one door opening with 2-3 bounces). Would this be possible?

Edit: To further clarify my point, I decided to add this part.

```
000000000000000000000000000000000000000000001111111111100000001111000111001010111111111111111111111
                                            |                                                     |
                                     HASS notified                        3 seconds after the signal
```
Basically, as soon as a change in the signal is detected, Home Assistant gets notified. After the change is detected, all other signals are ignored for a specific amount of time, let's say 3 seconds. After the three seconds, the Raspberry starts listening for changes again.

**Comment by thecode** [comment]
Denounce is used to eliminate hardware issues, in general every button or a switch has a short time which it's state may be unstable. However this events should be very short in time, as the example suggested 50ms, with 100ms being on the very high end.

Changing the algorithm as you suggested might help you but will not eliminate problems of noise or unstable contact which is what this is supposed to solve.

What you are looking can be solved in terms of a simple automation. The automation should catch the first state change and start a timer or ignore evens for certain amount of time.
You can also create a binary sensor which will implement what you suggested above using template.
https://www.home-assistant.io/integrations/template/#state-based-binary-sensor---washing-machine-running

I highly recommend to try HA [Community Forum](https://community.home-assistant.io) or [Discord chat server](https://discord.gg/c5DvZ4e) and try to get help how to do it with automation/template
